render.go

```diff
@@ -38,9 +39,18 @@ func (r *Render) Setup() {
 	}
 }
 
-func (r *Render) renderPrefix() {
+// getCurrentPrefix to get current prefix.
+// If live-prefix is enabled, return live-prefix.
+func (r *Render) getCurrentPrefix() string {
+	if prefix, ok := r.livePrefixCallback(); ok {
+		return prefix
+	}
+	return r.prefix
+}
+
+func (r *Render) renderPrefix(p string) {
 	r.out.SetColor(r.prefixTextColor, r.prefixBGColor, false)
-	r.out.WriteStr(r.prefix)
+	r.out.WriteStr(p)
```

Comment:
Change to `r.out.WriteStr(r.getCurrentPrefix())` and please remove a `p` argument.